core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -141,6 +141,26 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     this
   }
 
+  /** Set a boolean parameter */
```

Comment:
Rather than all these overloads, why not expose `set(key: String, value: Any)` since it is stored internally as a `String` from `toString()` anyway? This lets you add a generic `setIfMissing` too without so many methods. 
Would you also then change the code that calls `set` to not have to make it into a `String` in the caller? to gain the benefit that this enables.